orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartJenkinsJobTask.groovy

```diff
@@ -41,7 +50,41 @@ class StartJenkinsJobTask implements Task {
   TaskResult execute(@Nonnull StageExecution stage) {
     String master = stage.context.master
     String job = stage.context.job
-    String queuedBuild = buildService.build(master, job, stage.context.parameters)
-    TaskResult.builder(ExecutionStatus.SUCCEEDED).context([queuedBuild: queuedBuild]).build()
+
+    try {
+      Response igorResponse = buildService.build(master, job, stage.context.parameters)
+
+      if (igorResponse.getStatus() == HttpStatus.ACCEPTED.value()) {
+        log.info("build start for $master $job is pending, waiting")
+        return TaskResult.RUNNING
+      }
+
+      if (igorResponse.getStatus() == HttpStatus.OK.value()) {
+        String queuedBuild = igorResponse.body.in().text
+        return TaskResult
+            .builder(ExecutionStatus.SUCCEEDED)
+            .context([queuedBuild: queuedBuild])
+            .build()
+      }
+    }
```

Comment:
I think RetrofitError will be handled by RetrofitException handler which decides whether Retry should happen or not. May be better to leave it to that and not handle Retrofit error here.

Reply:
Might want to flag this as a `RetryableTask` now?

Reply:
> I think RetrofitError will be handled by RetrofitException handler which decides whether Retry should happen or not. May be better to leave it to that and not handle Retrofit error here.

It is true, it will. I am torn on this because since I know it can happen here and is retryable, i'd rather add a more specific error handler for clarity of logs/debugging. But I also recognize that this is code duplication... What do you think, @srekapalli ?

> Might want to flag this as a `RetryableTask` now?

I thought it is? Double checking

Reply:
> > I think RetrofitError will be handled by RetrofitException handler which decides whether Retry should happen or not. May be better to leave it to that and not handle Retrofit error here.
> 
> It is true, it will. I am torn on this because since I know it can happen here and is retryable, i'd rather add a more specific error handler for clarity of logs/debugging. But I also recognize that this is code duplication... What do you think, @srekapalli ?

May be log and re-throw it and not do any task status updates here.

Reply:
Looking more at this, I still prefer to keep it as is here. The reason being is that I don't want to rely on some  exception handler because, someone could (and probably should make this call a `POST` in which case it won't be retryable). I also considered wrapping and rethrowing a `SpinnakerException` with `isRetryable=true` but that creates a ton of noise in the logs (e.g. "failure running task, etc"). And i'd rather have the log clear and logic localized for when things do go wrong because it makes it easier to debug. What do you think?

Reply:
Sounds good.